### PR TITLE
Site transfers: Display notice on transfer success

### PR DESCRIPTION
**Pull request by kozer**
Closes https://github.com/Automattic/dotcom-forge/issues/2580

## Proposed Changes

This pr proposes adding a new success notice when the user successfully completes a site transfer

Backend: D112556-code

## Testing Instructions

1. Visit `/sites?site-transfer-confirm=true`
2. Ensure that you see the following:

success_transfer_notice

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/sites-dashboard/components/sites-dashboard.tsx [hunk 1]
```diff
@@ -351,3 +352,25 @@ function useShowSiteCreationNotice( allSites: SiteExcerptData[], newSiteID: numb
...
+	const { __ } = useI18n();
+	const dispatch = useDispatch();
+	const shownSiteTransferredNotice = useRef( false );
+
+	useEffect( () => {
+		if ( shownSiteTransferredNotice.current ) {
+			return;
+		}
+		const url = new URL( window.location.href );
+		if ( url.searchParams.get( 'transfer_complete' ) === 'true' ) {
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-dashboard.tsx]: Can we use `site-transfer-complete` for consistency with `site-transfer-confirm`?

File: client/sites-dashboard/components/sites-dashboard.tsx [hunk 2]
```diff
@@ -351,3 +352,25 @@ function useShowSiteCreationNotice( allSites: SiteExcerptData[], newSiteID: numb
...
+			shownSiteTransferredNotice.current = true;
+
+			dispatch( successNotice( __( 'Your blog transfer succeeded!' ), { duration: 8000 } ) );
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-dashboard.tsx]:
```suggestion
			dispatch( successNotice( __( 'Site transfer complete.' ), { duration: 8000 } ) );
```

@javierarce What do you think about this copy, for consistency with 'Post published' and 'SSH key added.'?

File: client/sites-dashboard/components/sites-dashboard.tsx (hunk 1 above)
Comment (kozer) [reply client/sites-dashboard/components/sites-dashboard.tsx]: I added it like that, as originally, it was `&msg=transfercomplete`. I wanted to kinda keep the same identifier. Do you think it makes more sense? If not, I can change it to `site-transfer-confirm`

**Update:**
Changed.

File: client/sites-dashboard/components/sites-dashboard.tsx (hunk 2 above)
Comment (kozer) [reply client/sites-dashboard/components/sites-dashboard.tsx]: Just FYI, again, as in my above comment states, I added it like this, to keep a "consistency" between `my-blogs` and this one.

File: client/sites-dashboard/components/sites-dashboard.tsx [hunk 3]
```diff
@@ -351,3 +352,25 @@ function useShowSiteCreationNotice( allSites: SiteExcerptData[], newSiteID: numb
...
+
+function useShowSiteTransferredNotice() {
+	const { __ } = useI18n();
+	const dispatch = useDispatch();
+	const shownSiteTransferredNotice = useRef( false );
+
+	useEffect( () => {
+		if ( shownSiteTransferredNotice.current ) {
+			return;
+		}
```
Comment (sejas) [inline client/sites-dashboard/components/sites-dashboard.tsx]: @kozer , Do we need this ref `shownSiteTransferredNotice` ?
I tried removing this guard, and the variable on my local machine and it works nice. It displays the notice only once.

**Review by sejas — COMMENTED** [review]
I tested it and works as described.

I added a suggestion because I think we can remove unnecessary code.

**Review by katinthehatsite — APPROVED** [review]
This works as expected for me 👍

**Review by katinthehatsite — APPROVED** [review]
This works as expected for me 👍

File: client/sites-dashboard/components/sites-dashboard.tsx (hunk 2 above)
Comment (katinthehatsite) [reply client/sites-dashboard/components/sites-dashboard.tsx]: I think in any case, we should use at least "Your site transfer succeeded!" to be consistent with the "site transfer" language that we are using everywhere else

Comment (kozer) [reply client/sites-dashboard/components/sites-dashboard.tsx]: I like that @katinthehatsite , I'll go with that!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8021644

Thank you @kozer for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8021644).

**Pull request merged**